drivers/homeless_summary_report/app/controllers/homeless_summary_report/warehouse_reports/reports_controller.rb

```diff
@@ -0,0 +1,166 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HomelessSummaryReport::WarehouseReports
+  class ReportsController < ApplicationController
+    include WarehouseReportAuthorization
+    include AjaxModalRails::Controller
+    include ArelHelper
+    include BaseFilters
+
+    before_action :require_can_access_some_version_of_clients!, only: [:details]
+    before_action :set_report, only: [:show, :destroy, :details]
+    before_action :set_variants, only: [:show, :details]
+    before_action :set_m1_fields, only: [:show, :details]
+    before_action :set_m2_fields, only: [:show]
+    before_action :set_m7_fields, only: [:show, :details]
```

Comment:
All of this logic should be moved into the report model.  Then the instance variables should be replaced with calls to those methods.